pkg/corerp/frontend/controller/applications/getgraph.go

```diff
@@ -0,0 +1,93 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package applications
+
+import (
+	"context"
+	"net/http"
+
+	v1 "github.com/radius-project/radius/pkg/armrpc/api/v1"
+	"github.com/radius-project/radius/pkg/corerp/datamodel"
+	"github.com/radius-project/radius/pkg/corerp/datamodel/converter"
+	"github.com/radius-project/radius/pkg/sdk"
+	"github.com/radius-project/radius/pkg/ucp/resources"
+
+	ctrl "github.com/radius-project/radius/pkg/armrpc/frontend/controller"
+	"github.com/radius-project/radius/pkg/armrpc/rest"
+)
+
+var _ ctrl.Controller = (*GetApplicationGraph)(nil)
+
+// GetApplicationGraph is the controller implementation to get application graph.
+type GetApplicationGraph struct {
+	ctrl.Operation[*datamodel.Application, datamodel.Application]
+	conn sdk.Connection
+}
+
+// NewGetApplicationGraph creates a new instance of the GetApplicationGraph controller.
+func NewGetApplicationGraph(opts ctrl.Options, UCPConn sdk.Connection) (ctrl.Controller, error) {
+	return &GetApplicationGraph{
+		ctrl.NewOperation(opts,
+			ctrl.ResourceOptions[datamodel.Application]{
+				RequestConverter:  converter.ApplicationDataModelFromVersioned,
+				ResponseConverter: converter.ApplicationDataModelToVersioned,
+			},
+		),
+		UCPConn,
+	}, nil
+}
+
+func (ctrl *GetApplicationGraph) Run(ctx context.Context, w http.ResponseWriter, req *http.Request) (rest.Response, error) {
+	sCtx := v1.ARMRequestContextFromContext(ctx)
+
+	// Request route for getGraph has name of the operation as suffix which should be removed to get the resource id.
+	// route id format: /planes/radius/local/resourcegroups/default/providers/Applications.Core/applications/corerp-resources-application-app/getGraph"
+	applicationID := sCtx.ResourceID.Truncate()
+	applicationResource, _, err := ctrl.GetResource(ctx, applicationID)
+	if err != nil {
+		return nil, err
+	}
+	if applicationResource == nil {
+		return rest.NewNotFoundResponse(sCtx.ResourceID), nil
+	}
+	// An application **MUST** have an environment id
+	environmentID, err := resources.Parse(applicationResource.Properties.Environment)
+	if err != nil {
+		return nil, err
+	}
+
+	clientOptions := sdk.NewClientOptions(ctrl.conn)
+
+	// get all resources in application scope
+	applicationResources, err := listAllResourcesByApplication(ctx, applicationID, clientOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	// get all resources in environment scope
+	environmentResources, err := listAllResourcesByEnvironment(ctx, environmentID, clientOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	if applicationResources == nil && environmentResources == nil {
```

Comment:
I had originally missed adding if applicationResources == nil && environmentResources == nil  clause. While local testing it turned out the app I was testing with had a deployment failure. In that case the getGraph API caused a crash. Therefore, I added this check, after which it was just a 200 OK with no body. 

@rynowak @ytimocin @vinayada1  I am not sure what error code would be appropriate here. "Internal Error", since we are dealing with an app that doesn't have resources? It's a bit confusing since the "error" happened earlier during the actual deployment.

Reply:
For now, I am keeping it as 200 OK with empty graph response from computeGraph function, please let me know if this should change.

Reply:
I think the issue mentions that we will ignore errors and return partial results in those cases. So this approach sounds okay to me.